Code/Engine/Core/Scripting/Implementation/ScriptComponent.cpp

```diff
@@ -0,0 +1,302 @@
+#include <Core/CorePCH.h>
+
+#include <Core/Scripting/ScriptComponent.h>
+#include <Core/Scripting/ScriptWorldModule.h>
+#include <Core/WorldSerializer/WorldReader.h>
+#include <Core/WorldSerializer/WorldWriter.h>
+
+// clang-format off
+EZ_BEGIN_COMPONENT_TYPE(ezScriptComponent, 1, ezComponentMode::Static)
+{
+  EZ_BEGIN_PROPERTIES
+  {
+    EZ_ACCESSOR_PROPERTY("UpdateInterval", GetUpdateInterval, SetUpdateInterval)->AddAttributes(new ezClampValueAttribute(ezTime::Zero(), ezVariant())),
+    EZ_ACCESSOR_PROPERTY("ScriptClass", GetScriptClassFile, SetScriptClassFile)->AddAttributes(new ezAssetBrowserAttribute("CompatibleAsset_ScriptClass")),
+    EZ_MAP_ACCESSOR_PROPERTY("Parameters", GetParameters, GetParameter, SetParameter, RemoveParameter)->AddAttributes(new ezExposedParametersAttribute("ScriptClass")),
+  }
+  EZ_END_PROPERTIES;
+  EZ_BEGIN_ATTRIBUTES
+  {
+    new ezCategoryAttribute("Scripting"),
+    new ezInDevelopmentAttribute(ezInDevelopmentAttribute::Phase::Alpha),
+  }
+  EZ_END_ATTRIBUTES;
+}
+EZ_END_DYNAMIC_REFLECTED_TYPE;
+// clang-format on
+
+ezScriptComponent::ezScriptComponent() = default;
+ezScriptComponent::~ezScriptComponent() = default;
+
+void ezScriptComponent::SerializeComponent(ezWorldWriter& stream) const
+{
+  SUPER::SerializeComponent(stream);
+  auto& s = stream.GetStream();
+
+  s << m_hScriptClass;
+  s << m_UpdateInterval;
+
+  ezUInt16 uiNumParams = static_cast<ezUInt16>(m_Parameters.GetCount());
+  s << uiNumParams;
+
+  for (ezUInt32 p = 0; p < uiNumParams; ++p)
+  {
+    s << m_Parameters.GetKey(p);
+    s << m_Parameters.GetValue(p);
+  }
+}
+
+void ezScriptComponent::DeserializeComponent(ezWorldReader& stream)
+{
+  SUPER::DeserializeComponent(stream);
+  // const ezUInt32 uiVersion = stream.GetComponentTypeVersion(GetStaticRTTI());
+  auto& s = stream.GetStream();
+
+  s >> m_hScriptClass;
+  s >> m_UpdateInterval;
+
+  ezUInt16 uiNumParams = 0;
+  s >> uiNumParams;
+  m_Parameters.Reserve(uiNumParams);
+
+  ezHashedString key;
+  ezVariant value;
+  for (ezUInt32 p = 0; p < uiNumParams; ++p)
+  {
+    s >> key;
+    s >> value;
+
+    m_Parameters.Insert(key, value);
+  }
+}
+
+void ezScriptComponent::Initialize()
+{
+  SUPER::Initialize();
+
+  if (m_hScriptClass.IsValid())
+  {
+    InstantiateScript(false);
+  }
+}
+
+void ezScriptComponent::Deinitialize()
+{
+  SUPER::Deinitialize();
+
+  ClearInstance(false);
+}
+
+void ezScriptComponent::OnActivated()
+{
+  SUPER::OnActivated();
+
+  CallScriptFunction(ezComponent_ScriptBaseClassFunctions::OnActivated);
```

Comment:
is this executed at editor time as well ?

Reply:
yes it is